src/go/rpk/pkg/cli/cmd/redpanda/admin/brokers/decommission.go

```diff
@@ -0,0 +1,86 @@
+package brokers
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/api/admin"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/redpanda"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newDecommissionBroker(fs afero.Fs) *cobra.Command {
+	var force bool
+	cmd := &cobra.Command{
+		Use:   "decommission [BROKER ID]",
+		Short: "Decommission the given broker",
+		Long: `Decommission the given broker.
+
+Decommissioning a broker removes it from the cluster.
+
+A decommission request is sent to every broker in the cluster, only the cluster
+leader handles the request.
+`,
+		Args: cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			broker, err := strconv.Atoi(args[0])
+			out.MaybeDie(err, "invalid broker %s: %v", args[0], err)
+			if broker < 0 {
+				out.Die("invalid negative broker id %v", broker)
+			}
+
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			cl, err := admin.NewClient(fs, cfg)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			if !force {
+				brokers, err := cl.Brokers(cmd.Context())
+				out.MaybeDie(err, "unable to get broker list: %v", err)
+
+				var b admin.Broker
+				var found bool
+				for _, br := range brokers {
+					if br.NodeID == broker {
+						b, found = br, true
+						break
```

Comment:
Per @bharathv's comment, it's more important for us to check the leader is a certain version, not the node that is down. We can't get the leader from this endpoint, but I think a decent fallback (per @bharathv's suggestion) is to ensure _every_ node in a cluster is v23.1+.

This is still _slightly_ racey: if a v22.3 comes back online and then becomes controller between these two commands, we run the risk, but I think that window is so small that it's ok